# <font color='blue'> Stop Words </font>

**A stop word is a commonly used word (such as “the”, “a”, “an”, “in”) that a search engine has been programmed to ignore, both when indexing entries for searching and when retrieving them as the result of a search query.** 

**Spacy has built in 305 words**

In [1]:
import spacy

In [2]:
nlp=spacy.load('en_core_web_sm')

In [4]:
print(nlp.Defaults.stop_words)

{'or', 'always', 'wherein', 'both', 'often', 'last', 'i', 'those', 'while', 'latter', 'your', '‘ve', 'next', 'since', 'became', 'thereafter', 'seemed', 'former', 'would', 'whom', 'along', 'thus', 'we', 'above', 'against', '’s', 'sixty', 'however', 'yours', 'nevertheless', 'hence', 'done', 'other', 'their', 'much', 'hundred', '’d', 'ever', 'one', 'him', '‘ll', '’m', 'anyway', 'am', 'will', 'several', 'amount', 'mine', 're', 'mostly', 'were', 'moreover', 'ours', "'d", 'please', 'whole', 'whence', 'first', 'show', 'some', 'third', 'could', 'indeed', 'from', 'with', 'beforehand', 'no', 'therein', "'s", 'alone', 'various', 'must', 'still', 'get', 'afterwards', 'fifteen', '’ve', 'are', 'to', 'regarding', 'seems', 'full', 'yourself', "'ve", 'whether', '‘m', 'among', 'empty', 'already', 'sometime', 'nobody', '‘d', 'each', 'at', 'top', 'through', 'can', 'take', 'but', 'under', 'noone', 'become', 'had', 'make', 'been', 'go', 'together', 'down', 'serious', 'upon', 'us', 'very', 'whoever', "n't", 

**These are the list of all default stop words presnet**

In [5]:
len(nlp.Defaults.stop_words)

326

###  <font color='purple'>Method to check if a word is stop word or not</font>

In [6]:
nlp.vocab['is'].is_stop

True

In [7]:
nlp.vocab['GFG'].is_stop

False

In [8]:
nlp.vocab['hello'].is_stop

False

In [11]:
nlp.vocab['i.e'].is_stop

False

**To add stop words of your own**

suppose we want to add i.e to indicate ' that is '

In [13]:
nlp.Defaults.stop_words.add('i.e')

In [15]:
nlp.vocab['i.e'].is_stop = True

In [16]:
len(nlp.Defaults.stop_words)

327

Previously it was 326 but now it has become 327

In [17]:
nlp.vocab['i.e'].is_stop

True

Therefore i.e has been added to out stop words dictionary

**Lets see how you would remove a stop word**

In [18]:
nlp.vocab['done'].is_stop

True

In [19]:
nlp.Defaults.stop_words.remove('done')

In [20]:
nlp.vocab['done'].is_stop = False

In [21]:
nlp.vocab['done'].is_stop

False

Therefore done has been removed from the stop words category